src/iexamine.cpp

```diff
@@ -1836,7 +1837,7 @@ void iexamine::kiln_full(player &, const tripoint &examp)
     }
 
     item result( "charcoal", calendar::turn.get_turn() );
-    result.charges = total_volume * char_type->ammo->def_charges / char_type->volume;
+    result.charges = total_volume / char_type->volume;
```

Comment:
`itype::volume` is now per single charge, so `total_volume / volume_per_single_charge` gives the number of charges that make up that volume. That's what `charges_per_volume` calculates.